### PR TITLE
feat(brokers): Add brokers for `mob_android_v1` and `mob_ios_v1`

### LINKED ISSUE #5029: Create signin context for standalone mobile fxa libs
We're creating an [Android](https://github.com/mcomella/FirefoxAccounts-android) and [iOS](https://github.com/liuche/FirefoxAccounts-ios) standalone fxa/Sync library. In the short term, these are to download read-only from Sync.

During the web flow sign in, we're currently using the context "fx_ios_v1" to initiate the [postMessage interface](https://github.com/mcomella/FirefoxAccounts-android/blob/master/accounts/src/main/res/raw/firefox_account_login.js#L17) (we modeled the login code after FF iOS).

We should create a new sign in context to differentiate these new libraries. I propose: 

"mob_[platform]_v1"

e.g. "mob_ios_v1".

fwiw, I'd expect these libraries to be ready for use within the next few weeks (we'll use fx_ios_v1 until this bug is complete).

CC @liuche.

### PR DESCRIPTION
**Pull request by shane-tomlinson**
fixes #5029

### CODE REVIEW

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: cfcc81826e681b4ff6d2de8e20e84e5843b6d782
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/mozilla/fxa-content-server/blob/master/CONTRIBUTING.md#git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

**Comment by shane-tomlinson** [comment]
This is ready for review, @mozilla/fxa-devs - r? @mcomella gave it a "WFM" in #5029

File: app/tests/spec/models/auth_brokers/fx-sync-channel.js [hunk 1]
```diff
@@ -251,18 +252,69 @@ define(function (require, exports, module) {
...
+        });
+      });
+
+      describe('browser supports `sendAfterSignInConfirmationPollNotice`', () => {
+        it('notifies the channel of login, does not halt the flow by default', () => {
+          broker.setCapability('sendAfterSignInConfirmationPollNotice', true);
+          return broker.afterSignInConfirmationPoll(account)
+            .then((result) => {
+              assert.isTrue(channelMock.send.calledOnce);
+              assert.isTrue(channelMock.send.calledWith('email_verified'));
```
Comment (shane-tomlinson) [inline app/tests/spec/models/auth_brokers/fx-sync-channel.js]: Ensure login data is sent too.

File: app/tests/spec/models/auth_brokers/fx-sync-channel.js [hunk 2]
```diff
@@ -251,18 +252,69 @@ define(function (require, exports, module) {
...
+        });
+      });
+
+      describe('browser supports `sendAfterSignUpConfirmationPollNotice`', () => {
+        it('notifies the channel of login, does not halt the flow by default', () => {
+          broker.setCapability('sendAfterSignUpConfirmationPollNotice', true);
+          return broker.afterSignUpConfirmationPoll(account)
+            .then((result) => {
+              assert.isTrue(channelMock.send.calledOnce);
+              assert.isTrue(channelMock.send.calledWith('email_verified'));
```
Comment (shane-tomlinson) [inline app/tests/spec/models/auth_brokers/fx-sync-channel.js]: Ensure login data is sent too.

File: app/scripts/models/auth_brokers/fx-sync-channel.js [hunk 3]
```diff
@@ -40,7 +41,9 @@ define(function (require, exports, module) {
     commands: null,
 
     defaultCapabilities: _.extend({}, proto.defaultCapabilities, {
-      sendChangePasswordNotice: true
+      sendAfterSignInConfirmationPollNotice: false,
+      sendAfterSignUpConfirmationPollNotice: false,
```
Comment (philbooth) [inline app/scripts/models/auth_brokers/fx-sync-channel.js]: A spot of bike-shedding here with my "ubiquitous language" hat on:

I thought we named sign-in *confirmation* to differentiate it from the *verification* step that happens after sign-up. Are we calling them both "confirmation" everywhere now?

File: app/scripts/models/auth_brokers/fx-sync-channel.js [hunk 4]
```diff
@@ -131,6 +134,30 @@ define(function (require, exports, module) {
         .then(() => proto.beforeSignUpConfirmationPoll.call(this, account));
     },
 
+    afterSignInConfirmationPoll (account) {
+      return proto.afterSignInConfirmationPoll.call(this, account)
+        .then((behavior) => {
+          if (this.hasCapability('sendAfterSignInConfirmationPollNotice')) {
+            const loginData = this._getLoginData(account);
+            return this.send(this.getCommand('EMAIL_VERIFIED'), loginData)
```
Comment (philbooth) [inline app/scripts/models/auth_brokers/fx-sync-channel.js]: Once upon a time we had plans to replace the current sign-in confirmation flow with other approaches, Google Authenticator etc. Would this command name seem weird if/when that happens?

**Comment by philbooth** [comment]
LGTM! r+

File: app/scripts/models/auth_brokers/fx-sync-channel.js (hunk 3 above)
Comment (shane-tomlinson) [inline app/scripts/models/auth_brokers/fx-sync-channel.js]: No, this is the semantics of the content server. "Confirm" occurs in the tab the user initiated the action. "verify" occurs in the verification tab.

File: app/scripts/models/auth_brokers/fx-sync-channel.js (hunk 4 above)
Comment (shane-tomlinson) [inline app/scripts/models/auth_brokers/fx-sync-channel.js]: > Once upon a time we had plans to replace the current sign-in confirmation flow with other approaches, Google Authenticator etc. Would this command name seem weird if/when that happens?

Possibly. We could just call it "VERIFIED" and ignore the verification method.

File: app/scripts/models/auth_brokers/fx-sync-channel.js (hunk 3 above)
Comment (shane-tomlinson) [inline app/scripts/models/auth_brokers/fx-sync-channel.js]: The above comment is not to mean that the semantics the content server uses are not totally messed up and confusing. The _are both_. I'd love help renaming these methods/concepts internally, every person that comes into the project is confused by the names & corresponding URL scheme.

File: app/scripts/models/auth_brokers/fx-sync-channel.js (hunk 4 above)
Comment (shane-tomlinson) [inline app/scripts/models/auth_brokers/fx-sync-channel.js]: Pulling [over the comment from the issue](https://github.com/mozilla/fxa-content-server/issues/5029#issuecomment-308397925):

@mcomella - if we changed the WebChannel message from fxaccounts:email_verified to fxaccounts:verified, would that cause havoc for you?

Comment (shane-tomlinson) [inline app/scripts/models/auth_brokers/fx-sync-channel.js]: @mcomella says "@shane-tomlinson Nope! Go for it."

**Pull request merged**